principal/event.go

```diff
@@ -348,6 +350,29 @@ func (s *Server) processAppProjectEvent(ctx context.Context, agentName string, e
 	return nil
 }
 
+// processClusterCacheInfoUpdateEvent processes the clusterCacheInfoUpdate event received from the agent.
+// It updates the cluster cache stats in the principal's cluster cache.
+func (s *Server) processClusterCacheInfoUpdateEvent(agentName string, ev *cloudevents.Event) error {
+	clusterInfo := &v1alpha1.ClusterInfo{}
```

Comment:
As above, IMHO our protocol should not hardcode embedded Argo CD objects. We should use our own objects (which copy the fields of Argo CD's)